src/main/java/org/sunbird/cb/hubservices/serviceimpl/ConnectionService.java

```diff
@@ -30,23 +32,32 @@ public class ConnectionService implements IConnectionService {
 	@Autowired
 	INodeService nodeService;
 	@Override
-	public Response upsert(ConnectionRequest request) {
-		Response response = null;
+	public Response upsert(ConnectionRequest request, String updateOperation) {
+		Response response = new Response();
+		Node from = new Node();
+		Node to = new Node();
 		if(validateRequest(request)) {
-			Node to = new Node(request.getUserIdFrom());
-			Node from = new Node(request.getUserIdTo());
+			if(updateOperation.equalsIgnoreCase(Constants.ADD_OPERATION)) {
+				from = new Node(request.getUserIdFrom());
+				to = new Node(request.getUserIdTo());
+			}
+			else if(updateOperation.equalsIgnoreCase(Constants.UPDATE_OPERATION))
+			{
+				to = new Node(request.getUserIdFrom());
+				from = new Node(request.getUserIdTo());
+			}
```

Comment:
This is bad programming... there is no need to declare default constructor and then instantiate the objects with it...